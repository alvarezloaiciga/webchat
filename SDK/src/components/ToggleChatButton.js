// @flow
/** @jsx h */
import {h} from 'preact';
import {getStyle} from 'Common/QuiqOptions';
import {getQuiqOptions} from 'Globals';
import {displayModes} from 'Common/Constants';
import InvitationBlurb from './InvitationBlurb';
import styled, {css} from 'preact-emotion';

export type ToggleChatButtonProps = {
  onClick: () => void,
  open: boolean,
  invitationBlurbText: string,
};

const StyledInvitationBubble = styled(InvitationBlurb)`
  position: fixed;
  right: 90px;
  bottom: 20px;
`;

const ToggleChatButtonStyle = css`
  z-index: 99999;
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  box-sizing: border-box;
  bottom: 24px;
  right: 24px;
  width: 60px;
  height: 60px;
  border: none;
  border-radius: 100%;
  color: #fff;
  fill: #fff;
  cursor: pointer;
  transition: 0.15s ease-in-out all;
  box-shadow: rgba(0, 0, 0, 0.117647) 0px 1px 6px, rgba(0, 0, 0, 0.117647) 0px 1px 4px;

  // Leave these in, some customer sites have default stylings for buttons that try to override these
  margin: 0;
  padding: 0;

  &:hover {
    box-shadow: rgba(0, 0, 0, 0.156863) 0px 3px 10px, rgba(0, 0, 0, 0.227451) 0px 3px 10px;
  }

  &:focus {
    outline: none;
  }

  svg {
    fill: #fff;
    flex: 1 1 auto;
  }
`;

const ToggleChatButton = ({onClick, open, invitationBlurbText}: ToggleChatButtonProps) => {
  const {styles, colors, displayMode} = getQuiqOptions();

  return (
    <div>
      <button
        className={`ToggleChatButton ${ToggleChatButtonStyle}`}
        onClick={onClick}
        style={getStyle(styles.ToggleChatButton, {backgroundColor: colors.primary})}
      >
        <svg
          style={getStyle(styles.ToggleChatButtonIcon)}
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          viewBox="0 0 24 24"
        >
          {open && displayMode !== displayModes.UNDOCKED ? (
            <path d="M23 20.168l-8.185-8.187 8.185-8.174-2.832-2.807-8.182 8.179-8.176-8.179-2.81 2.81 8.186 8.196-8.186 8.184 2.81 2.81 8.203-8.192 8.18 8.192z" />
          ) : (
            <path d="M12 1c-6.628 0-12 4.573-12 10.213 0 2.39.932 4.591 2.427 6.164l-2.427 5.623 7.563-2.26c9.495 2.598 16.437-3.251 16.437-9.527 0-5.64-5.372-10.213-12-10.213z" />
          )}
        </svg>
      </button>
      {invitationBlurbText && (
        <StyledInvitationBubble
          text={invitationBlurbText}
          style={getStyle(styles.InvitationBlurb, {
            backgroundColor: 'white',
            shadowColor: '#B2B2B2',
          })}
        />
      )}
    </div>
  );
};

export default ToggleChatButton;
