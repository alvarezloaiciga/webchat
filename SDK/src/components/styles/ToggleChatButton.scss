.ToggleChatButton {
  z-index: 99999;
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  box-sizing: border-box;
  bottom: 24px;
  right: 24px;
  width: 60px;
  height: 60px;
  border: none;
  border-radius: 100%;
  color: #fff;
  fill: #fff;
  cursor: pointer;
  transition: 0.15s ease-in-out all;
  box-shadow: rgba(0, 0, 0, 0.117647) 0px 1px 6px, rgba(0, 0, 0, 0.117647) 0px 1px 4px;

  // Leave these in, some customer sites have default stylings for buttons that try to override these
  margin: 0; 
  padding: 0;

  &:hover {
    box-shadow: rgba(0, 0, 0, 0.156863) 0px 3px 10px, rgba(0, 0, 0, 0.227451) 0px 3px 10px;
  }

  &:focus {
    outline: none;
  }

  svg {
    fill: #fff;
    flex: 1 1 auto;
  }
}
