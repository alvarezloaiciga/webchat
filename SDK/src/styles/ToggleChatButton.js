export default {
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  position: 'fixed',
  boxSizing: 'border-box',
  bottom: '24px',
  right: '24px',
  width: '60px',
  height: '60px',
  border: 'none',
  borderRadius: '100%',
  color: '#fff',
  fill: '#fff',
  cursor: 'pointer',
  transition: '0.15s ease-in-out all',
  boxShadow: 'rgba(0, 0, 0, 0.117647) 0px 1px 6px, rgba(0, 0, 0, 0.11 7647) 0px 1px 4px',
};
