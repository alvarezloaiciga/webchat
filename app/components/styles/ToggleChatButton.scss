#quiqWebChat {
  .ToggleChatButton {
    position: fixed;
    bottom: 24px;
    right: 24px;
    width: 60px;
    height: 60px;
    border: none;
    border-radius: 100%;
    color: #fff;
    cursor: pointer;
    transition: 0.15s ease-in-out all;
    box-shadow: rgba(0, 0, 0, 0.117647) 0px 1px 6px, rgba(0, 0, 0, 0.117647) 0px 1px 4px;

    &:hover {
      box-shadow: rgba(0, 0, 0, 0.156863) 0px 3px 10px, rgba(0, 0, 0, 0.227451) 0px 3px 10px;
    }

    &:focus {
      outline: none;
    }

    svg {
      fill: #fff;
    }
  }
}
