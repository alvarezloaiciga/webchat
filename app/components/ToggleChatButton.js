// @flow

import React from 'react';
import quiqOptions, {getStyle} from 'Common/QuiqOptions';
import ChatBubbleIcon from './ChatBubbleIcon';
import {connect} from 'react-redux';
import {isMobile} from 'Common/Utils';
import type {ChatState} from 'Common/types';
import './styles/ToggleChatButton.scss';

export type ToggleChatButtonProps = {
  toggleChat: () => void,
  chatContainerHidden: boolean,
};

const {colors, styles} = quiqOptions;

export const ToggleChatButton = ({toggleChat, chatContainerHidden}: ToggleChatButtonProps) => {
  const buttonStyle = getStyle(styles.ToggleChatButton, {backgroundColor: colors.primary});
  const iconStyle = getStyle(styles.ToggleChatButtonIcon);

  return (
    <button style={buttonStyle} onClick={toggleChat} className="ToggleChatButton">
      {!chatContainerHidden && !isMobile()
        ? <svg
            style={iconStyle}
            xmlns="http://www.w3.org/2000/svg"
            width="24"
            height="24"
            viewBox="0 0 24 24"
          >
            <path d="M23 20.168l-8.185-8.187 8.185-8.174-2.832-2.807-8.182 8.179-8.176-8.179-2.81 2.81 8.186 8.196-8.186 8.184 2.81 2.81 8.203-8.192 8.18 8.192z" />
            X
          </svg>
        : <ChatBubbleIcon style={iconStyle} />}
    </button>
  );
};

export default connect((state: ChatState) => ({
  chatContainerHidden: state.chatContainerHidden,
}))(ToggleChatButton);
