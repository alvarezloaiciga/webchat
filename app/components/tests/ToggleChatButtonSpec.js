// @flow
import React from 'react';
import type { ToggleChatButtonProps } from '../ToggleChatButton';
import ToggleChatButton from '../ToggleChatButton';
import { shallow } from 'enzyme';
import type { ShallowWrapper } from 'enzyme';


describe('ToggleChatButton component', () => {
  let wrapper:ShallowWrapper;
  let testProps:ToggleChatButtonProps;
  let render: () => void;

  beforeEach(() => {
    testProps = {
      toggleChat: jest.fn(),
      chatOpen: false,
    };
    render = () => {
      wrapper = shallow(<ToggleChatButton {...testProps}/>);
    };
  });

  describe('rendering', () => {
    beforeEach(() => {
      render();
    });

    it('renders with default props', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('open', () => {
    it('switches svgs', () => {
      testProps.chatOpen = true;
      render();

      expect(wrapper).toMatchSnapshot();
    });
  });
});
