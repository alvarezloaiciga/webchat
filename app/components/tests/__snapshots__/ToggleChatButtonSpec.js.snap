// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleChatButton component open switches svgs 1`] = `
<button
  className="ToggleChatButton"
  onClick={[Function]}
  style={
    Object {
      "background": "#333",
    }
  }
>
  <svg
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M23 20.168l-8.185-8.187 8.185-8.174-2.832-2.807-8.182 8.179-8.176-8.179-2.81 2.81 8.186 8.196-8.186 8.184 2.81 2.81 8.203-8.192 8.18 8.192z"
    />
    X
  </svg>
</button>
`;

exports[`ToggleChatButton component rendering renders with default props 1`] = `
<button
  className="ToggleChatButton"
  onClick={[Function]}
  style={
    Object {
      "background": "#333",
    }
  }
>
  <svg
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 1c-6.628 0-12 4.573-12 10.213 0 2.39.932 4.591 2.427 6.164l-2.427 5.623 7.563-2.26c9.495 2.598 16.437-3.251 16.437-9.527 0-5.64-5.372-10.213-12-10.213z"
    />
    O
  </svg>
</button>
`;
